### DISCUSSION TITLE
Automatically determine container format based on codec

### DISCUSSION THREAD

**Discussion by benniekiss** [PyAV-Org/PyAV#1529] [1. Help] [answered]
Hi all,

I was wondering if there is a way to programmatically determine the output container format based on the input media codec when transcoding?

My use case is that I am trying to extract the audio streams from various types of media and store these in an `io.BytesIO` buffer object. In doing so, I have to specify the container format since one cannot be auto-detected from the output filepath (since there is none).

I've considered creating a conversion table storing `codec -> format` key/values, but I wanted to check first if there was a more adaptable/automatic way to figure this out.

Here is the code I am using:

```python 
buffer = io.BytesIO()
with av.open("./media.mp4", "r") as container:
    a_stream = container.streams.audio[0]
    codec = a_stream.codec.name
    rate = a_stream.rate
    layout = a_stream.layout.name

    ## NEEDS IMPLEMENTED
    format = "some_container_format"

    with av.open(buffer, format=format, "w") as o_container:
        o_stream = o_container.add_stream(codec, layout=layout, rate=rate)
        for frame in container.decode(a_stream):
            for packet in o_stream.encode(frame):
                o_container.mux(packet)
        for packet in o_stream.encode(None):
            o_container.mux(packet)
```

**Comment by WyattBlue** [comment, marked as answer]
There isn't one. That's because you can store all audio codecs in the matroska format (mkv)

**Reply by benniekiss** [reply]
That makes sense! I didnt know that about mkv containers. Thank you
